### PR TITLE
Avoid requests with no shop code

### PR DESCRIPTION
**Pull request by MrGoumX**

### CODE REVIEW

File: spec/session_spec.coffee [hunk 1]
```diff
@@ -579,3 +579,13 @@ describe 'Session', ->
...
                   expect(@biskoto.get(@basic_sa_cookie_name).version).to.equal @settings.cookies.version
                   done()
+
+      context 'when the shop_code is not present', ->
+        beforeEach ->
+          @init = =>
+            sa('session', 'connect', '')
+            @instance = new @session(@plugins_manager).run()
+
+        it 'does not call XDomainEngine', ->
```
Comment (itholoenos) [inline spec/session_spec.coffee:589]: Can we check also that plugins_manager.notify is not called?

File: src/session.coffee [hunk 2]
```diff
@@ -30,6 +30,8 @@ define [
           @_extractAnalyticsSessionOnCreate()
 
         connect: (shop_code)->
+          return console?.warn?('Connect called with invalid shop code') unless shop_code
```
Comment (mrwhizzy) [inline src/session.coffee:33]: should we use `Connect called without a shop code` instead of `Connect called with invalid shop code`?

Comment (itholoenos) [reply src/session.coffee:33]: +1 ^

File: spec/session_spec.coffee (hunk 1 above)
Comment (mrwhizzy) [reply spec/session_spec.coffee:589]: agreed, an extra check wouldn't hurt

**Review by mrwhizzy — APPROVED** [review]
+1 LGTM!

**Review by itholoenos — APPROVED** [review]
+1 LGTM!

File: src/session.coffee [hunk 3]
```diff
@@ -30,6 +30,8 @@ define [
           @_extractAnalyticsSessionOnCreate()
 
         connect: (shop_code)->
+          return console?.warn?('Connect called without a shop code') unless shop_code
```
Comment (harrisred) [inline src/session.coffee:33]: We could easily catch the whitespace cases as well: `skroutz_analytics('session', 'connect', '  ');`. Although I see no such requests in the logs.

```coffeescript
return console?.warn?('Connect called without a shop code') if !shop_code || shop_code.trim?() == ''
```

Comment (harrisred) [reply src/session.coffee:33]: If we decide to move forward with this we should add an extra test. Also we could add another test when the shop_code is null.

File: src/plugins_manager.coffee [hunk 4]
```diff
@@ -21,6 +21,8 @@ define [
     @param [Object, String] data Any additional data required by the plugins
     ###
     notify: (action, data) ->
+      return unless @session?.shop_code
```
Comment (harrisred) [inline src/plugins_manager.coffee:24]: I believe this is redundant. We should ✂️ . `notify` will [never be called](https://github.com/skroutz/analytics.js/blob/master/src/session.coffee#L40) due to the [session connect safe guard](https://github.com/skroutz/analytics.js/pull/83/commits/098e21ee87748fa7ee43ae0fed96abab6391a3f1#diff-620774fd73e3a52fa36c5c3beac74b331e6b3491e3d28a6702eab919ed0a63eaR33).

File: src/session.coffee (hunk 3 above)
Comment (itholoenos) [reply src/session.coffee:33]: I agree with this, we should trim the `shop_code` and then check it. We should update the specs to test for `null`, `''` and `' '`.

Comment (mrwhizzy) [reply src/session.coffee:33]: I would be tempted to take this a step further and suggest a basic validation along the lines of `SA-[0-9]{4,}-[0-9]{4,}`

Comment (harrisred) [reply src/session.coffee:33]: Discussed offline. We decided to not trim the shop code.

File: src/plugins_manager.coffee (hunk 4 above)
Comment (itholoenos) [reply src/plugins_manager.coffee]: done!

**Review by harrisred — APPROVED** [review]
LGTM 👍
